functions/v2/label_gce_instance/main.py

```
@@ -0,0 +1,89 @@
+    auth_info = payload.get('authenticationInfo', dict())
+    creator = auth_info.get('principalEmail')
+
+    # Get relevant VM instance details from the cloudevent's `subject` property
+    # Example value:
+    #   compute.googleapis.com/projects/<PROJECT_ID>/zones/<ZONE_ID>/instances/<INSTANCE_NAME>
+    instance_params = cloudevent['subject'].split('/')
+
+    # Validate data
+    if not creator or not instance_params or len(instance_params) != 7:
+        # This is not something retries will fix, so don't throw an Exception
+        # (Thrown exceptions trigger retries *if* you enable retries in GCF.)
+        print('ERROR: Invalid `principalEmail` and/or CloudEvent `subject`.')
+        return
+
+    instance_project = instance_params[2]
+    instance_zone = instance_params[4]
+    instance_name = instance_params[6]
+
+    # Format the 'creator' parameter to match GCE label validation requirements
+    creator = re.sub('\\W', '_', creator.lower())
+
+    # Get the newly-created VM instance's label fingerprint
+    # This is required by the Compute Engine API to prevent duplicate labels
+    instance = instances_client.get(
+        project=instance_project,
+        zone=instance_zone,
+        instance=instance_name
+    )
+
+    # Construct API call to label the VM instance with its creator
+    request_init = {
+        'project': instance_project,
+        'zone': instance_zone,
+        'instance': instance_name
+    }
+    request_init['instances_set_labels_request_resource'] = \
+        compute.InstancesSetLabelsRequest(
+            label_fingerprint=instance.label_fingerprint,
+            labels={'creator': creator}
+        )
+    request = compute.SetLabelsInstanceRequest(request_init)
+
+    # Perform instance-labeling API call
+    try:
+        instances_client.set_labels(request)
+        print(f'Labelled VM instance {instance_name} with creator: {creator}')
+    except Exception as e:
```

I see the note about swallowing the exception but am concerned that catching the general exception is detrimental - is there a more specific one that can be caught instead?

---

I'm not 100% positive this is the right one, but `GoogleAPIError` [seems like a better candidate](https://googleapis.dev/python/google-api-core/latest/exceptions.html#google.api_core.exceptions.GoogleAPICallError)?

I went ahead and switched to that.

---

Definitely a better choice than just Exception 😁 ! Without much knowledge about the product I can't tell ya too much more than that but I like this better